### ISSUE TITLE
[Bug]: Incognito context page won't authenticate to proxy after Monterey update?

### ISSUE THREAD

**Issue by DavideValdo** [puppeteer/puppeteer#7827]
### Bug description

Doesn't happen in the remote Ubuntu 18 server executing the same code

Steps to reproduce the problem:

1. Update to macOS Monterey 
2. Run a node script that launches puppeteer like this (dotenv is working correctly, double-checked):
```
const browser = 
await puppeteer.launch({
      args: [
        `--proxy-server=${process.env.PROXY_URL}`,
        "--no-sandbox",
        "--user-agent=Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/96.0.4664.45 Safari/537.36",
      ],
      headless: true,
    });
    
const context = await browser.createIncognitoBrowserContext();

const page = await context.newPage();

// Disable images altogether
await page.setRequestInterception(true);
page.on("request", (req) => {
    if (req.resourceType() === "image") {
    req.abort();
    return;
    }

    req.continue();
});

await page.authenticate({
    username: process.env.USERNAME,
    password: process.env.PASSWORD,
});

```
3. Been trying both headless and headful modes, when running headful if I try to open a new tab in the same browser window, I get prompted for proxy login credentials

Any logs I'm missing? The above code is wrapped in a `try/catch` block but won't throw exceptions.

### Puppeteer version

remote: 10.4.0, locally: 10.4.0 (tried 12.0.1)

### Node.js version

remote: 14.7.5, local: 16.13.1

### npm version

local: 8.2.0

### What operating system are you seeing the problem on?

macOS

**Comment by roudhentelnagoy** [comment]
Incognito is not using the proxy server declared when you launch the browser. You need to pass the proxyServer when you create new incognito browser context

`await browser.createIncognitoBrowserContext({
      proxyServer: 'http://host:port'
    })`

**Comment by Cooya** [comment]
This bug is fixed in the [version 13.5.0](https://github.com/puppeteer/puppeteer/releases/tag/v13.5.0).
